File: DSA/DSA04010 DÃY CON LIÊN TIẾP CÓ TỔNG LỚN NHẤT.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n, max = -101, stored = 0;
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        int temp;
        cin >> temp;

        stored += temp;
        if (stored < temp)
        {
            stored = temp;
        }
        
        if (stored > max)
        {
            max = stored;
        }
    }

    cout << max;

    cout << endl;
}

int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);

    int test;
    cin >> test;
    while (test--)
    {
        submit();
    }
    

    return 0;
}

File: DSA_python/DSA04010 DÃY CON LIÊN TIẾP CÓ TỔNG LỚN NHẤT.py
def submit():
    n = int(input())
    max, stored = -101, 0
    A = list(map(int, input().split()))

    for i in A:
        stored += i
        if stored < i: stored = i
        if stored > max: max = stored

    print(max)

test = int(input())
for _ in range(test):
    submit()
